lib/spack/spack/compilers/msvc.py

```diff
@@ -305,6 +305,10 @@ def platform_toolset_ver(self):
         vs22_toolset = Version(toolset_ver) > Version("142")
         return toolset_ver if not vs22_toolset else "143"
 
+    @property
+    def visual_studio_version(self):
+        return re.search(r"[0-9]{4}", self.cc).group(0)
```

Comment:
As written, this seems like this could intersect with the `version_regex`: would it be possible to add some of the "surrounding structure" (i.e. static strings in the vicinity of this sequence)? Or is that subject to change?

Reply:
This is parsing the path to the compiler executable, whereas the version regex is parsing the version output string. There are components of that path that are fairly static, but since this is an install path, multiple components are subject to change across msvc versions and install locations. That said, I think there's at least some fixed path components we can rely on...